py/objexcept.c

```diff
@@ -114,7 +114,15 @@ void mp_obj_exception_print(const mp_print_t *print, mp_obj_t o_in, mp_print_kin
         } else if (o->args->len == 1) {
             // try to provide a nice OSError error message
             if (o->base.type == &mp_type_OSError && MP_OBJ_IS_SMALL_INT(o->args->items[0])) {
-                const char* msg = mp_errno_to_str(o->args->items[0]);
+                const compressed_string_t* common = mp_common_errno_to_str(o->args->items[0]);
+                const char* msg;
+                char decompressed[50];
```

Comment:
not 51 for final null?

Reply:
50 is arbitrary since we don't know the length yet. That's why its checked below against the actual length (which includes NULL). It has to be declared in this scope so it survives down to the print call below.

Reply:
50 is ok, we subtract 1 here
https://github.com/adafruit/circuitpython/pull/1121/files#diff-3c0b0b9ab47034d1e9ce2494287f5b68R70